classes/class-sensei-teacher.php

```diff
@@ -1405,4 +1412,67 @@ public function limit_teacher_edit_screen_post_types( $wp_query ) {
 
     } // end limit_teacher_edit_screen_post_types()
 
+
+    /**
+     * Sensei_Teacher::restrict_posts_menu_page()
+     *
+     * Remove the Posts menu page for teachers and restrict access to it.
+     * We have to do this because we give teachers the 'edit_posts' cap
+     * so they can 'moderate_comments' as well.
+     *
+     * @since 1.8.7
+     * @access public
+     * @parameters void
+     * @return void
+     */
+
+    public function restrict_posts_menu_page() {
+
+        global $pagenow, $typenow;
+
+        if (current_user_can('teacher')) {
+
+            remove_menu_page('edit.php');
+
+            if ($pagenow == "edit.php" || $pagenow == "post-new.php") {
+
+                if ($typenow == '' || $typenow == 'post' || $typenow == 'page') {
+
+                    wp_redirect(admin_url());
```

Comment:
@mahangu could we make it so that  the user get's a no permissions notice instead?  So just a blank screen with no permissions message like WP itself shows users. Redirecting to admin without a notice may be very confusing.